tests/foreman/ui_airgun/test_domain.py

```diff
@@ -0,0 +1,335 @@
+        })
+        assert session.domain.search(name), (
+            "Unable to find domain '{}' after creating"
+            .format(name)
+        )
+
+
+@run_only_on('sat')
+@upgrade
+def test_positive_delete(session):
+    """Delete a domain
+
+    :id: 07c1cc34-4569-4f04-9c4a-2842821a6977
+
+    :expectedresults: Domain is deleted
+
+    :CaseImportance: Critical
+    """
+    with session:
+        name = DOMAIN % gen_string('alphanumeric')
+        session.domain.create({
+            'domain.dns_domain': name,
+            'domain.full_name': name,
+        })
+        session.domain.delete(name)
+        assert not session.domain.search(name), (
+            "Deleted domain '{}' still exists in UI"
+            .format(name)
+        )
+
+
+@run_only_on('sat')
+@upgrade
+@parametrize('new_name', **valid_domain_names(length=5))
+def test_positive_update(session, new_name):
+    """Update a domain with name and description
+
+    :id: 25ff4a1d-3ca1-4153-be45-4fe1e63f3f16
+
+    :expectedresults: Domain is updated
+
+    :CaseImportance: Critical
+    """
+    with session:
+        old_name = DOMAIN % gen_string('alphanumeric')
+        session.domain.create({
+            'domain.dns_domain': old_name,
+            'domain.full_name': old_name,
+        })
+        session.domain.update(old_name, {'domain.dns_domain': new_name})
+        assert session.domain.search(new_name), (
+            "Unable to find domain '{}' after changing name (old name: {})"
+            .format(new_name, old_name)
+        )
+
+
+@run_only_on('sat')
+@parametrize('name', **invalid_domain_names())
+def test_negative_create_with_invalid_name(session, name):
+    """Try to create domain and use whitespace, blank, tab symbol or
+    too long string of different types as its name value
+
+    :id: 5a8ba1a8-2da8-48e1-8b2a-96d91161bf94
+
+    :expectedresults: Domain is not created
+
+    :CaseImportance: Critical
+    """
+    with session:
+        with pytest.raises(AssertionError) as context:
+            session.domain.create({
+                'domain.dns_domain': name,
+                'domain.full_name': name,
+            })
+        assert 'errors present' in str(context.value)
+
+
+@run_only_on('sat')
+@tier2
+def test_positive_set_parameter(session):
+    """Set parameter name and value for domain
+
+    :id: a05615de-c9e5-4784-995c-b2fe2a1dfd3e
+
+    :expectedresults: Domain is updated
+
+    :CaseLevel: Integration
+    """
+    update_values = {
+        'parameters.params': [
+            {
+                'name': gen_string('alpha'),
+                'value': gen_string('alpha')
+            }
+        ]
+    }
+    with session:
+        name = DOMAIN % gen_string('alphanumeric')
+        session.domain.create({
+            'domain.dns_domain': name,
+            'domain.full_name': name,
+        })
+        session.domain.update(name, update_values)
+
+
+@run_only_on('sat')
+@tier2
+def test_positive_set_parameter_long(session):
```

Comment:
not sure that previous test case is needed. We can make one with previous and verify only '255' values